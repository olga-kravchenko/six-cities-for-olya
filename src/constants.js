const CityNames = {
  PARIS: `Paris`,
  COLOGNE: `Cologne`,
  BRUSSELS: `Brussels`,
  AMSTERDAM: `Amsterdam`,
  HAMBURG: `Hamburg`,
  DUSSELDORF: `Dusseldorf`,
};

const SortingType = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

const DEFAULT_CITY = CityNames.PARIS;
const DEFAULT_SORTING_TYPE = SortingType.POPULAR;
const DEFAULT_STATE = false;

const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer/:id`,
  PAGE_NOT_FOUND: `/page-not-found`,
};

const AxiosRoute = {
  OFFERS: `/hotels`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  NEARBY: `/nearby`,
  FAVORITE: `/favorite`,
  COMMENTS: `/comments`,
};

const HttpCode = {
  OK: 200,
  NO_FOUND: 404,
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
};

export {
  CityNames,
  SortingType,
  DEFAULT_CITY,
  DEFAULT_SORTING_TYPE,
  DEFAULT_STATE,
  AppRoute,
  AxiosRoute,
  HttpCode
};
