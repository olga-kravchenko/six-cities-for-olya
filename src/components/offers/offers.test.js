import React from "react";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import Offers from "./offers";
import {DEFAULT_SORTING_TYPE} from "../../constants";

describe(`Offers should render correctly`, () => {
  it(`Offers should render correctly on Main page`, () => {
    const extendedOffers = [
      {id: 1},
      {id: 3},
      {id: 2},
    ];
    const mockStore = configureStore({});
    const store = mockStore({
      SORTING_TYPE: {sortingType: DEFAULT_SORTING_TYPE},
    });
    const {container} = render(
        <Provider store={store}>
          <Offers offers={extendedOffers} pageType="cities"/>
        </Provider>);
    expect(container).toMatchSnapshot();
  });

  it(`Offers should render correctly on Favorite page`, () => {
    const extendedOffers = [
      {id: 1},
      {id: 3},
      {id: 2},
    ];
    const mockStore = configureStore({});
    const store = mockStore({
      SORTING_TYPE: {sortingType: DEFAULT_SORTING_TYPE},
    });
    const {container} = render(
        <Provider store={store}>
          <Offers offers={extendedOffers} pageType="favorites"/>
        </Provider>);
    expect(container).toMatchSnapshot();
  });
});
