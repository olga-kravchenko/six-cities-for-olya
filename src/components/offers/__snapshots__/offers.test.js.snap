// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard should render correctly on Main page 1`] = `
<div>
  <div
    class="cities__places-list places__list tabs__content"
  >
    <article
      class="cities__place-card place-card"
      id="1"
    >
      <div
        class="cities__image-wrapper place-card__image-wrapper"
      >
        <a>
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            width="260"
          />
        </a>
      </div>
      <div
        class="place-card__info "
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button "
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/1"
          />
        </h2>
        <p
          class="place-card__type"
        />
      </div>
    </article>
    <article
      class="cities__place-card place-card"
      id="3"
    >
      <div
        class="cities__image-wrapper place-card__image-wrapper"
      >
        <a>
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            width="260"
          />
        </a>
      </div>
      <div
        class="place-card__info "
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button "
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/3"
          />
        </h2>
        <p
          class="place-card__type"
        />
      </div>
    </article>
    <article
      class="cities__place-card place-card"
      id="2"
    >
      <div
        class="cities__image-wrapper place-card__image-wrapper"
      >
        <a>
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            width="260"
          />
        </a>
      </div>
      <div
        class="place-card__info "
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button "
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/2"
          />
        </h2>
        <p
          class="place-card__type"
        />
      </div>
    </article>
  </div>
</div>
`;

exports[`Offers should render correctly Offers should render correctly on Favorite page 1`] = `
<div>
  <div
    class="cities__places-list places__list tabs__content"
  >
    <article
      class="favorites__card place-card"
      id="1"
    >
      <div
        class="favorites__image-wrapper place-card__image-wrapper"
      >
        <a>
          <img
            alt="Place image"
            class="place-card__image"
            height="110"
            width="150"
          />
        </a>
      </div>
      <div
        class="place-card__info favorites__card-info"
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button "
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/1"
          />
        </h2>
        <p
          class="place-card__type"
        />
      </div>
    </article>
    <article
      class="favorites__card place-card"
      id="3"
    >
      <div
        class="favorites__image-wrapper place-card__image-wrapper"
      >
        <a>
          <img
            alt="Place image"
            class="place-card__image"
            height="110"
            width="150"
          />
        </a>
      </div>
      <div
        class="place-card__info favorites__card-info"
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button "
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/3"
          />
        </h2>
        <p
          class="place-card__type"
        />
      </div>
    </article>
    <article
      class="favorites__card place-card"
      id="2"
    >
      <div
        class="favorites__image-wrapper place-card__image-wrapper"
      >
        <a>
          <img
            alt="Place image"
            class="place-card__image"
            height="110"
            width="150"
          />
        </a>
      </div>
      <div
        class="place-card__info favorites__card-info"
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button "
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/2"
          />
        </h2>
        <p
          class="place-card__type"
        />
      </div>
    </article>
  </div>
</div>
`;

exports[`Offers should render correctly Offers should render correctly on Main page 1`] = `
<div>
  <div
    class="cities__places-list places__list tabs__content"
  >
    <article
      class="cities__place-card place-card"
      id="1"
    >
      <div
        class="cities__image-wrapper place-card__image-wrapper"
      >
        <a>
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            width="260"
          />
        </a>
      </div>
      <div
        class="place-card__info "
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button "
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/1"
          />
        </h2>
        <p
          class="place-card__type"
        />
      </div>
    </article>
    <article
      class="cities__place-card place-card"
      id="3"
    >
      <div
        class="cities__image-wrapper place-card__image-wrapper"
      >
        <a>
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            width="260"
          />
        </a>
      </div>
      <div
        class="place-card__info "
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button "
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/3"
          />
        </h2>
        <p
          class="place-card__type"
        />
      </div>
    </article>
    <article
      class="cities__place-card place-card"
      id="2"
    >
      <div
        class="cities__image-wrapper place-card__image-wrapper"
      >
        <a>
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            width="260"
          />
        </a>
      </div>
      <div
        class="place-card__info "
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button "
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/2"
          />
        </h2>
        <p
          class="place-card__type"
        />
      </div>
    </article>
  </div>
</div>
`;
